




<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN"
"http://www.w3.org/TR/REC-html40/loose.dtd">
<html>
<head>
<title>Repository - markup - qemu: qemu/pc-bios/bios.diff</title>
<link rel="stylesheet" href="/viewcvs-doc/styles.css" type="text/css">
</head>
<body>
<table width="100%" border=0 cellpadding=0 cellspacing=0>
<tr><td valign="bottom">
<h3>Savannah CVS Surfing


- project <a href="http://savannah.nongnu.org/projects/qemu">qemu</a></div>


</h3>
<div class="vc_navheader">
<a href="/viewcvs/?root=qemu#dirlist">[qemu]</a> / <a href="/viewcvs/qemu/?root=qemu#dirlist">qemu</a> / <a href="/viewcvs/qemu/pc-bios/?root=qemu#dirlist">pc-bios</a> / bios.diff
</div>



<code>cvs -d:pserver:anonymous@cvs.sv.gnu.org:/sources/qemu co qemu/pc-bios/bios.diff</code>



</td>
<td align="right">
<a href="http://savannah.gnu.org/"><img src="http://savannah.gnu.org/images/transparent.theme/floating.png" alt="Savannah" border="0" width="148" height="125"></a><br />
<a href="/viewcvs-doc/help_rootview.html">ViewCVS and CVS Help</a>
</td></tr>
</table>
<div class="vc_summary">
File: <a href="/viewcvs/?root=qemu#dirlist">[qemu]</a> / <a href="/viewcvs/qemu/?root=qemu#dirlist">qemu</a> / <a href="/viewcvs/qemu/pc-bios/?root=qemu#dirlist">pc-bios</a> / bios.diff
(<a href="/viewcvs/*checkout*/qemu/pc-bios/bios.diff?rev=1.18&root=qemu"><b>download</b></a>)

<br>

Revision: <b>1.18</b>,
<i>Thu Feb  8 22:17:34 2007 UTC</i> (7 weeks, 4 days ago) by <i>bellard</i>

<br>Branch: <b>MAIN</b>


<br>CVS Tags: <b>HEAD</b>




<br>Changes since <b>1.17: +24 -2 lines</b>




<pre class="vc_log">reset rombios32 area
</pre>

</div>
<pre>Index: rombios.c
===================================================================
RCS file: /cvsroot/bochs/bochs/bios/rombios.c,v
retrieving revision 1.174
diff -u -w -r1.174 rombios.c
--- rombios.c	17 Oct 2006 16:48:05 -0000	1.174
+++ rombios.c	8 Feb 2007 21:57:48 -0000
@@ -9472,6 +9472,14 @@
   mov eax, #0x00040000
   call eax
 
+  ;; reset the memory (some boot loaders such as syslinux suppose 
+  ;; that the memory is set to zero)
+  mov edi, #0x00040000
+  mov ecx, #0x40000 / 4
+  xor eax, eax
+  rep 
+    stosd
+
   ;; return to 16 bit protected mode first
   db 0xea
   dd rombios32_10
Index: rombios.h
===================================================================
RCS file: /cvsroot/bochs/bochs/bios/rombios.h,v
retrieving revision 1.3
diff -u -w -r1.3 rombios.h
--- rombios.h	3 Oct 2006 20:27:30 -0000	1.3
+++ rombios.h	8 Feb 2007 21:57:48 -0000
@@ -19,7 +19,7 @@
 //  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
 
 /* define it to include QEMU specific code */
-//#define BX_QEMU
+#define BX_QEMU
 
 #ifndef LEGACY
 #  define BX_ROMBIOS32     1
Index: rombios32.c
===================================================================
RCS file: /cvsroot/bochs/bochs/bios/rombios32.c,v
retrieving revision 1.8
diff -u -w -r1.8 rombios32.c
--- rombios32.c	3 Oct 2006 20:27:30 -0000	1.8
+++ rombios32.c	8 Feb 2007 21:57:48 -0000
@@ -852,6 +852,11 @@
     int ioapic_id, i, len;
     int mp_config_table_size;
 
+#ifdef BX_QEMU
+    if (smp_cpus &lt;= 1)
+        return;
+#endif
+
 #ifdef BX_USE_EBDA_TABLES
     mp_config_table = (uint8_t *)(ram_size - ACPI_DATA_SIZE - MPTABLE_MAX_SIZE);
 #else
</pre>

<hr noshade>
<address>Send suggestions and report system problems to the <a href="https://savannah.gnu.org/support/?group=administration">Savannah Hackers</a>.</address>
</body></html>

